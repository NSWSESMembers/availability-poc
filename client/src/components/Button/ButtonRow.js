import React from 'react';
import PropTypes from 'prop-types';
import { TouchableOpacity, View } from 'react-native';
import ButtonText from './ButtonText';

const ButtonRow = ({ title, description, onPress, showIcon, showIconNoPress }) => (
  <View>
    {onPress === undefined ? (
      <ButtonText icon={showIconNoPress} title={title} description={description} />
    ) : (
      <TouchableOpacity onPress={onPress}>
        <ButtonText icon={showIcon} title={title} description={description} />
      </TouchableOpacity>
    )}
  </View>
);

ButtonRow.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string,
  onPress: PropTypes.func,
  showIcon: PropTypes.bool,
  showIconNoPress: PropTypes.bool,
};

ButtonRow.defaultProps = {
  showIcon: false,
  showIconNoPress: false,
};

export default ButtonRow;
