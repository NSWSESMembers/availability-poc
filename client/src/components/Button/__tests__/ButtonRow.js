import React from 'react';
import renderer from 'react-test-renderer';

import ButtonRow from '../ButtonRow';

const noop = () => {};

test('renders correctly', () => {
  const tree1 = renderer.create(
    <ButtonRow
      title="title"
      description="description"
      onPress={noop}
      showIcon
      showIconNoPress
    />,
  ).toJSON();
  expect(tree1).toMatchSnapshot();

  const tree2 = renderer.create(
    <ButtonRow
      title="title"
      description="description"
      showIcon
      showIconNoPress
    />,
  ).toJSON();
  expect(tree2).toMatchSnapshot();
});
