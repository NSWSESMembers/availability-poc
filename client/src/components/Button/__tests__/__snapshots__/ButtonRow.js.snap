// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 5,
          "paddingTop": 5,
        }
      }
    >
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 11,
              "fontWeight": "900",
              "paddingTop": 5,
            }
          }
        >
          TITLE
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "paddingBottom": 5,
            }
          }
        >
          description
        </Text>
      </View>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#EEE",
              "fontSize": 40,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders correctly 2`] = `
<View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 5,
        "paddingTop": 5,
      }
    }
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 11,
            "fontWeight": "900",
            "paddingTop": 5,
          }
        }
      >
        TITLE
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "paddingBottom": 5,
          }
        }
      >
        description
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#EEE",
            "fontSize": 40,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
